codegen/src/main/java/software/amazon/awssdk/codegen/model/intermediate/MemberModel.java

```diff
@@ -296,33 +299,14 @@ public MemberModel withXmlNameSpaceUri(String xmlNameSpaceUri) {
 
     public String getSetterDocumentation() {
         StringBuilder docBuilder = new StringBuilder();
-        docBuilder.append(getSetterDoc());
-
-        if ("java.nio.ByteBuffer".equals(
-                this.getGetterModel().getReturnType())) {
-
-            docBuilder.append("<p>")
-                    .append(LF)
-                    .append("AWS SDK for Java performs a Base64 " +
-                            "encoding on this field before sending this request to AWS " +
-                            "service by default. " +
-                            "Users of the SDK should not perform Base64 " +
-                            "encoding on this field.")
-                    .append(LF)
-                    .append("</p>")
-                    .append(LF);
-
-            docBuilder.append("<p>")
-                    .append(LF)
-                    .append("Warning: ByteBuffers returned by the SDK are mutable. " +
-                            "Changes to the content or position of the byte buffer will be " +
-                            "seen by all objects that have a reference to this object. " +
-                            "It is recommended to call ByteBuffer.duplicate() or " +
-                            "ByteBuffer.asReadOnlyBuffer() before using or reading from the buffer. " +
-                            "This behavior will be changed in a future major version of the SDK.")
-                    .append(LF)
-                    .append("</p>")
-                    .append(LF);
+
+        docBuilder.append(StringUtils.isNotBlank(documentation) ? documentation : DEFAULT_SETTER.replace("%s", name) + "\n");
+
+        if (ByteBuffer.class.getName().equals(this.getGetterModel().getReturnType())) {
+            appendParagraph(docBuilder, "The AWS SDK for Java performs a Base64 encoding on this field before sending this "
```

Comment:
That's fair. I could be less specific and specify that we will 'encode' this field before sending it, but that might be ambiguous enough to be inaccurate if some service comes along that actually requires some kind of encoding we don't do.